aws/nss_aws_login

```diff
@@ -0,0 +1,101 @@
+#
+
+profile_name="$1"
+if [ -z "$profile_name" ]; then
+  echo "usage: $(basename $0) PROFILE_NAME"
+  exit 1
+fi
+
+if ! aws configure list-profiles | grep $profile_name >/dev/null; then
+  echo "ERROR: Invalid AWS PROFILE_NAME specified"
+  exit 1
+fi
+
+echo "## Performing AWS Login with the profile $profile_name."
+echo "##"
+
+echo "## Step 1: aws sso logout"
+aws sso logout
+
+echo "## Step 2: aws sso login --profile $profile_name"
+aws sso login --profile $profile_name
+
+echo
+echo "## Step 3: make $profile_name the default profile"
+
+ssh_cache_file=$(find ~/.aws/sso/cache -maxdepth 1 -type f -name "[0-9]*.json" -exec grep -l accessToken {} \;)
+if [ -z "$ssh_cache_file" ]; then
+  echo "ERROR: Cannot find SSO file - have you run 'aws sso login --profile $profile_name' first?"
+  exit 1
+fi
+
+CREDS=~/.aws/credentials
+if [ ! -e $CREDS ]; then
+  echo "## Creating an empty credentials file"
+  mkdir -p $(dirname $CREDS)
+  touch $CREDS
+  chmod 600 $CREDS
+  echo "[default]" > $CREDS
+  echo "aws_access_key_id=tmp" >> $CREDS
+  echo "aws_secret_access_key=tmp" >> $CREDS
+  echo "aws_session_token=tmp" >> $CREDS
+else
+  echo "## Making backup copy of existing credentials file"
+  cp $CREDS ${CREDS}.bak
+fi
+# set -xv   # this line will enable debug
+echo "## Getting Account and Role Name from config"
+account_id=$(aws configure get sso_account_id --profile $profile_name)
+role_name=$(aws configure get sso_role_name --profile $profile_name)
+region=$(aws configure get region --profile $profile_name)
+
+# set the default region in config
+aws configure set region $region
+
+token=$(cat $ssh_cache_file | jq -r '.accessToken')
+
+# if credentials have expired (e.g. overnight) this commmand will fail, thus the check here
+echo "## Getting SSO role credentials"
+
+credentials=$(aws sso get-role-credentials --role-name $role_name --account-id $account_id --access-token $token --region $region)
+if [ -z "$credentials" ]; then
+  echo
+  echo "ERROR: Have you run 'aws sso login --profile $profile_name' first?"
+  exit 1
+fi
+
+# set +xv  # this line will enable debug
+# at this point we should have everything we need and can get the variables
+access_key=$(echo $credentials | jq -r '.roleCredentials.accessKeyId')
+access_secret=$(echo $credentials | jq -r '.roleCredentials.secretAccessKey')
+session_token=$(echo $credentials | jq -r '.roleCredentials.sessionToken')
+
+# awk magic to indiscriminately replace lines 2-4 in the credentials file
+cat $CREDS | \
+  awk -v access_key=$access_key '{ if(NR==2) print "aws_access_key_id=" access_key; else print $0}' | \
+  awk -v access_secret=$access_secret '{ if(NR==3) print "aws_secret_access_key=" access_secret; else print $0}' | \
+  awk -v session_token=$session_token '{if(NR==4) print "aws_session_token=" session_token; else print $0}' > ${CREDS}.new
+mv ${CREDS}.new $CREDS
+
+echo "## Validating AWS Identity with the following:"
+echo "Account ID : $account_id"
+echo "Profile    : $profile_name"
```

Comment:
Updated variable name and output label from Role and $role to Profile and $profile_name